audio/descriptors.c

```diff
@@ -42,81 +47,235 @@ __code __at(DSCR_AREA) struct usb_descriptors code_descriptors = {
+            .wTerminalType      = UAC_OUTPUT_TERMINAL_STREAMING,
+            .bAssocTerminal     = 0,
+            /* Connected to the input terminal */
+            .bSourceID          = 1,
+            .iTerminal          = 0,
+        },
+        .streaming0 = {
             .bLength            = USB_DT_INTERFACE_SIZE,
             .bDescriptorType    = USB_DT_INTERFACE,
-            .bInterfaceNumber   = 0,
+            .bInterfaceNumber   = 1,
             .bAlternateSetting  = 0,
+            .bNumEndpoints      = 0,
+            .bInterfaceClass    = USB_CLASS_AUDIO,
+            .bInterfaceSubClass = USB_SUBCLASS_AUDIOSTREAMING,
+            .bInterfaceProtocol = 0,
+            .iInterface         = 0,
+        },
+        .streaming1 = {
+            .bLength            = USB_DT_INTERFACE_SIZE,
+            .bDescriptorType    = USB_DT_INTERFACE,
+            .bInterfaceNumber   = 1,
+            .bAlternateSetting  = 1,
             .bNumEndpoints      = 1,
             .bInterfaceClass    = USB_CLASS_AUDIO,
             .bInterfaceSubClass = USB_SUBCLASS_AUDIOSTREAMING,
-            .bInterfaceProtocol = 0, // Must be zero
-            .iInterface         = USB_STRING_INDEX(2),
+            .bInterfaceProtocol = 0,
+            .iInterface         = 0,
+        },
+        .streamheader = {
+            .bLength            = UAC_DT_AS_HEADER_SIZE,
+            .bDescriptorType    = USB_DT_CS_INTERFACE,
+            .bDescriptorSubtype = UAC_AS_GENERAL,
+            .bTerminalLink      = 2,
+            .bDelay             = 1,
+            .wFormatTag         = UAC_FORMAT_TYPE_I_PCM,
+        },
+        .format = {
+            .bLength            = UAC_FORMAT_TYPE_I_DISCRETE_DESC_SIZE(1),
+            .bDescriptorType    = USB_DT_CS_INTERFACE,
+            .bDescriptorSubtype = UAC_FORMAT_TYPE,
+            .bFormatType        = UAC_FORMAT_TYPE_I,
+            .bNrChannels        = 2,
+            /* Bytes per audio subframe */
+            .bSubframeSize      = 2,
+            .bBitResolution     = 16,
+            /* Frequencies supported */
+            .bSamFreqType       = 1,
+            /* 8000Hz, little endian */
+            .tSamFreq[0]        = { 0x40, 0x1F, 0x00 },
         },
         .endpoints = {
             {
                 .bLength            = USB_DT_ENDPOINT_AUDIO_SIZE,
                 .bDescriptorType    = USB_DT_ENDPOINT,
-                .bEndpointAddress   = USB_ENDPOINT_NUMBER(0x2) | USB_DIR_OUT,
-                .bmAttributes       = 0x1, // Isynchronous endpoint
+                .bEndpointAddress   = USB_ENDPOINT_NUMBER(8) | USB_DIR_IN,
+                .bmAttributes       = (USB_ENDPOINT_XFER_ISOC | USB_ENDPOINT_SYNC_NONE),
                 .wMaxPacketSize     = 512,
-                .bInterval          = 1,
+                .bInterval          = 4,
                 .bRefresh           = 0,
                 .bSynchAddress      = 0,
             },
         },
+        .isoendpoint = {
```

Comment:
Why is the `isoendpoint` unused? Can you add a comment here about why?

Reply:
It is used...? Describes the ISO streaming

Reply:
The `/* Unused */` comment in the middle confused me....

Reply:
Clarified